darts/models/forecasting/conformal_models.py

```diff
@@ -0,0 +1,1858 @@
+        )
+        return self
+
+    def predict(
+        self,
+        n: int,
+        series: Union[TimeSeries, Sequence[TimeSeries]] = None,
+        past_covariates: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+        future_covariates: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+        num_samples: int = 1,
+        verbose: bool = False,
+        predict_likelihood_parameters: bool = False,
+        show_warnings: bool = True,
+        **kwargs,
+    ) -> Union[TimeSeries, Sequence[TimeSeries]]:
+        """Forecasts calibrated quantile intervals (or samples from calibrated intervals) for `n` time steps after the
+        end of the `series`.
+
+        It is important that the input series for prediction correspond to a calibration set - a set different to the
+        series that the underlying forecasting `model` was trained on.
+
+        Since it is a probabilistic model, you can generate forecasts in two ways:
+
+        - Predict the calibrated quantile intervals directly: Pass parameters `predict_likelihood_parameters=True`, and
+          `num_samples=1` to the forecast method.
+        - Predict stochastic samples from the calibrated quantile intervals: Pass parameters
+          `predict_likelihood_parameters=False`, and `num_samples>>1` to the forecast method.
+
+        Under the hood, the simplified workflow to produce one calibrated forecast/prediction for every step in the
+        horizon `n` is as follows (note: `cal_length` and `cal_stride` can be set at model creation):
+
+        - Extract a calibration set: The calibration set for each conformal forecast is automatically extracted from
+          the past of your input series relative to the forecast start point. The number of calibration examples
+          (forecast errors / non-conformity scores) to consider can be defined at model creation
+          with parameter `cal_length`. Note that when using `cal_stride>1`, a longer history is required since
+          the calibration examples are generated with stridden historical forecasts.
+        - Generate historical forecasts on the calibration set (using the forecasting model) with a stride `cal_stride`.
+        - Compute the errors/non-conformity scores (specific to each conformal model) on these historical forecasts
+        - Compute the quantile values from the errors / non-conformity scores (using our desired quantiles set at model
+          creation with parameter `quantiles`).
+        - Compute the conformal prediction: Using these quantile values, add calibrated intervals to (or adjust the
+          existing intervals of) the forecasting model's predictions.
+
+        Parameters
+        ----------
+        n
+            Forecast horizon - the number of time steps after the end of the series for which to produce predictions.
+        series
+            A series or sequence of series, representing the history of the target series whose future is to be
```

Comment:
If I understand correctly, if the calibration series overlaps or is contained in the training set, this may lead to overly optimistic prediction intervals. Should we add this disclaimer?

Reply:
Sounds good 👍